packages/client-node/src/result_set.ts

```diff
@@ -35,34 +38,64 @@ export class ResultSet implements BaseResultSet<Stream.Readable> {
 
     validateStreamFormat(this.format)
 
-    let decodedChunk = ''
+    let data: Buffer
+    let incompleteRow: Buffer | undefined
     const toRows = new Transform({
       transform(
         chunk: Buffer,
-        encoding: BufferEncoding,
+        _encoding: BufferEncoding,
         callback: TransformCallback
       ) {
-        decodedChunk += chunk.toString()
+        let currentDataSize
+        // The buffer was already allocated
+        // x2 chunk size should be enough for any leftover incomplete rows
+        if (data !== undefined) {
+          let chunkOffset
+          if (incompleteRow !== undefined) {
+            chunkOffset = incompleteRow.length
+            currentDataSize = chunkOffset + chunk.length
+            data.fill(incompleteRow, 0, chunkOffset)
+          } else {
+            chunkOffset = 0
+            currentDataSize = chunk.length
+          }
+          data.fill(chunk, chunkOffset, currentDataSize)
+          data.fill(0, currentDataSize, data.length)
```

Comment:
Maybe there is a more efficient way to work with these buffers. What is more interesting is that this implementation is 5-7% slower than the main branch (bugged) one. I was expecting it to be the other way around.